bot_code/modelHelpers/actions/action_factory.py

```diff
@@ -3,52 +3,108 @@
     LOSS_ABSOLUTE_DIFFERENCE
 from bot_code.modelHelpers.actions.split_action_handler import SplitActionHandler
 
-default_scheme = [[('steer', (-1, 1.5, .5)), ('pitch', (-1, 1.5, .5)), ('roll', (-1, 1.5, .5))],
-                  [('throttle', (-1, 2, 1)), ('jump', (0, 2, 1)), ('boost', (0, 2, 1)), ('handbrake', (0, 2, 1))],
-                  [('yaw', 'steer')]]
+from collections import namedtuple
 
-super_split_scheme = [[('throttle', (-1, 1.5, .5)), ('steer', (-1, 1.5, .5)),
-                       ('yaw', (-1, 1.5, .5)), ('pitch', (-1, 1.5, .5)), ('roll', (-1, 1.5, .5))],
-                      [('jump', (0, 2, 1)), ('boost', (0, 2, 1)), ('handbrake', (0, 2, 1))],
-                      []]
+'''
+ranges:
+  A range is a configuration object which holds information about its name, possible values and loss function.
+  Currently represented as a tuple:
+  - Name (string)
+  - Arguments to np.arange (tuple of (startInclusive, endExclusive, step))
+  - Optional: The loss function (one of the LOSS_* constants)
+combo_scheme:
+  Seems to be similar to ranges. TODO: figure out how this works in more detail. Loss functions seem to never be specified.

```

Comment:
Combo takes a list of ranges and combines them into a single action using permutations.

Example Boost, Jump, Handbrake
All the permutations would be internally represented by a single number that the model will output.  
False, False, True = 1
False, False, False = 0